karapace/protobuf/serialization.py

```diff
@@ -0,0 +1,316 @@
+    if options.HasField("php_class_prefix"):
+        result.append(OptionElement("php_class_prefix", OptionElement.Kind.STRING, options.php_class_prefix))
+    if options.HasField("php_namespace"):
+        result.append(OptionElement("php_namespace", OptionElement.Kind.STRING, options.php_namespace))
+    if options.HasField("php_generic_services"):
+        result.append(OptionElement("php_generic_services", OptionElement.Kind.BOOLEAN, options.php_generic_services))
+    if options.HasField("php_metadata_namespace"):
+        result.append(OptionElement("php_metadata_namespace", OptionElement.Kind.STRING, options.php_metadata_namespace))
+    if options.HasField("ruby_package"):
+        result.append(OptionElement("ruby_package", OptionElement.Kind.STRING, options.ruby_package))
+    return result
+
+
+def deserialize(schema_b64: str) -> ProtoFileElement:
+    serialized_pb = base64.b64decode(schema_b64, validate=True)
+    proto = google.protobuf.descriptor_pb2.FileDescriptorProto()
+    proto.ParseFromString(serialized_pb)
+    imports: list[TypeName] = []
+    public_imports: list[TypeName] = []
+    for index, dep in enumerate(proto.dependency):
+        if index in proto.public_dependency:
+            public_imports.append(TypeName(dep))
+        else:
+            imports.append(TypeName(dep))
+    types: list[TypeElement] = []
+    for enumtype in proto.enum_type:
+        types.append(_deserialize_enum(enumtype))
+    for msgtype in proto.message_type:
+        types.append(_deserialize_msg(msgtype))
+    options: list[OptionElement] = _deserialize_options(proto.options)
+
+    return ProtoFileElement(
+        DEFAULT_LOCATION,
+        package_name=PackageName(proto.package),
+        syntax=Syntax(proto.syntax),
+        imports=imports,
+        public_imports=public_imports,
+        types=types,
+        options=options,
+    )
+
+
+_LABEL_MAP = MappingProxyType(
+    {
+        Field.Label.OPTIONAL: google.protobuf.descriptor_pb2.FieldDescriptorProto.LABEL_OPTIONAL,
+        Field.Label.REQUIRED: google.protobuf.descriptor_pb2.FieldDescriptorProto.LABEL_REQUIRED,
+        Field.Label.REPEATED: google.protobuf.descriptor_pb2.FieldDescriptorProto.LABEL_REPEATED,
+    }
+)
+
+
+def _serialize_field_label(label: Field.Label) -> google.protobuf.descriptor_pb2.FieldDescriptorProto.Label.ValueType:
+    if label not in _LABEL_MAP:
+        raise NotImplementedError(f"Unsupported field label {label}")
+    return _LABEL_MAP[label]
+
+
+def _serialize_field(field: FieldElement) -> google.protobuf.descriptor_pb2.FieldDescriptorProto:
+    d = google.protobuf.descriptor_pb2.FieldDescriptorProto()
+    if field.label is not None:
+        d.label = _serialize_field_label(field.label)
+        if field.label == Field.Label.OPTIONAL:
+            d.proto3_optional = True
+    else:
+        d.label = _serialize_field_label(Field.Label.OPTIONAL)
+    if _REVERSE_TYPE_MAP.get(field.element_type) is not None:
```

Comment:
maybe `field.element_type in _REVERSE_TYPE_MAP`? or you explicitly may find `None` as value of a key in that dict?